File: problems/461fe85c-4e7d-41a5-97ba-cd8d526aa632/go/checker.go
package main

import "log"

func runChecker(testcases []*Testcase) *RunResult {
	runResult := &RunResult{
		IsPassed: true,
	}

	for _, testcase := range testcases {
		input := testcase.Input
		expected := testcase.Expected
		gotValue, stdout, err := captureStdout(isValid, input)
		if err != nil {
			log.Fatal(err)
		}

		got := gotValue[0].Bool()

		if expected != got {

			runResult.FailedTestcases = append(runResult.FailedTestcases, &FailedTestcase{
				Id:       testcase.Id,
				Input:    testcase.Input,
				Expected: testcase.Expected,
				Got:      got,
				StdOut:   stdout,
			})
			runResult.IsPassed = false
		}
	}

	return runResult
}

File: problems/defc7c02-4255-47be-9812-5a9c2468fadd/javascript/checker.js
const { FailedTestcase, RunResult, captureStdout } = require('./backend_checker.js');
const { integerToRoman } = require('./solution.js');

function runChecker(testcases) {
    const runResult = new RunResult();
    
    for (const testcase of testcases) {
        const input = testcase.input;
        const expected = testcase.expected;
        
        const capture = captureStdout(() => integerToRoman(input));
        const got = capture.result;
        
        if (expected !== got) {
            const failedTestcase = new FailedTestcase(
                testcase.id,
                testcase.input,
                testcase.expected,
                got,
                capture.stdout
            );
            
            runResult.failedTestcases.push(failedTestcase);
            runResult.isPassed = false;
        }
    }
    
    return runResult;
}

module.exports = { runChecker };
